tests/serverpod_test_server/lib/src/endpoints/json_protocol.dart

```diff
@@ -0,0 +1,36 @@
+import 'dart:async';
+
+import 'package:serverpod/serverpod.dart';
+import 'package:serverpod_test_server/src/generated/protocol.dart';
+
+class JsonProtocolEndpoint extends Endpoint {
+  Future<ScopeServerOnlyField> getJsonForProtocol(
+    Session session,
+  ) async {
+    return ScopeServerOnlyField(
```

Comment:
Suggestion: Add a value to `serverOnlyScope` on the first level as well and validate that we don't pass that in the body.

Same for the streaming case.